kale/embed/deepdta.py

```diff
@@ -0,0 +1,103 @@
+        embedding_dim (int): Dimensions of embedding labels/categories.
+        sequence_length (int): Max length of input sequence.
+        num_kernels (int): Number of kernels (filters).
+        kernel_length (int): Length of kernel (filter).
+    """
+
+    def __init__(self, num_embeddings, embedding_dim, sequence_length, num_kernels, kernel_length):
+        super(DeepDTAEncoder, self).__init__()
+        self.embedding = nn.Embedding(num_embeddings + 1, embedding_dim)
+        self.conv1 = nn.Conv1d(in_channels=sequence_length, out_channels=num_kernels, kernel_size=kernel_length)
+        self.conv2 = nn.Conv1d(in_channels=num_kernels, out_channels=num_kernels * 2, kernel_size=kernel_length)
+        self.conv3 = nn.Conv1d(in_channels=num_kernels * 2, out_channels=num_kernels * 3, kernel_size=kernel_length)
+        self.gmp = nn.AdaptiveMaxPool1d(output_size=1)
+
+    def forward(self, x):
+        x = self.embedding(x)
+        x = F.relu(self.conv1(x))
+        x = F.relu(self.conv2(x))
+        x = F.relu(self.conv3(x))
+        x = self.gmp(x)
+        x = x.squeeze(2)
+        return x
+
+
+class DrugGCNEncoder(nn.Module):
+    r"""
+    The GraphDTA's GCNs encoder module, which comprises three graph convolutional layers and one full connected layer.
+    The model is a variant of DeepDTA and is applied to encode drug molecule graph information. the original paper
+    is  `"GraphDTA: Predicting drug–target binding affinity with graph neural networks"
+    <https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btaa921/5942970>`_ .
+
+    Args:
+        in_channel (int): Size of each input node.
+        out_channel (int): Size of each output node.
+        dropout (float): Dropout rate (probability of an element to be zeroed in after full connected layer).
+    """
+    def __init__(self, in_channel=78, out_channel=128, dropout=0.2):
+        super(DrugGCNEncoder, self).__init__()
+        self.conv1 = GCNConv(in_channel, in_channel)
+        self.conv2 = GCNConv(in_channel, in_channel * 2)
+        self.conv3 = GCNConv(in_channel * 2, in_channel * 4)
+        self.fc = nn.Linear(in_channel * 4, out_channel)
+        self.relu = nn.ReLU()
+        self.dropout = nn.Dropout(dropout)
+
+    def forward(self, x, edge_index, batch):
+        x = self.relu(self.conv1(x, edge_index))
+        x = self.relu(self.conv2(x, edge_index))
+        x = self.relu(self.conv3(x, edge_index))
+        x = gmp(x, batch)
+        x = self.fc(x)
+        x = self.dropout(x)
+        return x
+
+
+class MLPDecoder(nn.Module):
+    r"""
+    The MLP decoder module, which comprises four full connected neutral network. It's a common decoder for decoding
+    drug-target encoding information.
+
+    Args:
+        in_dim (int): Dimensions of input feature.
+        hidden_dim (int): Dimensions of hidden layers.
```

Comment:
You used dimensions (plural) here but the type is int, which means a scalar so it should be the dimension rather than dimensions.